src/separateLater/Sentinel1/Metadata/Sentinel1Metadata.jl

```diff
@@ -0,0 +1,558 @@
+end
+
+"""
+GeolocationGrid
+
+returns structure of GeolocationGrid from metadata in .xml
+
+
+"""
+Base.@kwdef struct GeolocationGrid
+    lines::Vector{Int64}
+    samples::Vector{Int64}
+    latitude::Vector{Float64}
+    longitude::Vector{Float64}
+    azimuthTime::Vector{Dates.DateTime}
+    slantRangeTime::Vector{Float64}
+    elevationAngle::Vector{Float64}
+    incidenceAngle::Vector{Float64}
+    height::Vector{Float64}
+end
+
+
+"""
+MetaDataSentinel1:
+    Metadata structure for the Sentinel-1 satellite
+
+    
+Example:
+    slcMetadata = MetaDataSentinel1(metaDict)
+
+    Input:
+        metaDict: xml file.
+
+    can be accesed as, e.g., 
+    slcMetadata.product.radarFrequency --> 5.40500045433435e9::Float64
+    slcMetadata.header.swath --> 1::Int
+    slcMetadata.header.mode --> "IW"::String
+    slcMetadata.header.polarisation --> "VH"::String
+
+
+"""
+
+Base.@kwdef struct MetaDataSentinel1
+    header::Header
+    product::ProductInformation
+    image::ImageInformation
+    swath::SwathTiming
+    burst::Burst
+    geolocation::GeolocationGrid
+end
+
+
+
+
+##################
+### Constructors
+##################
+
+""""
+MetaDataSentinel1
+
+    Constucture for the MetaDataSentinel1 structure. 
+    It takes a Sentinel-1 single swath metafile in .xml format and constructs the metadata structure using the individual sub-structures in the metadata.
+    The individual sub-structures in the metadata are:
+    - Header
+    - ProductInformation
+    - ImageInformation
+    - SwathTiming
+    - Burst
+    - GeolocationGrid
+
+    Input:
+        xmlFile[string]: path of swath specific metadata in xml.format.
+
+    output:
+        MetaDataSentinel1[structure of MetaDataSentinel1]: Strucutre with all Sentinel-1 metadata for a swath.
+    
+
+"""
+function MetaDataSentinel1(xmlFile::String)::MetaDataSentinel1
+
+    metaDict = getDictofXml(xmlFile)
+
+
+    swathtiming = SwathTiming(metaDict)
+    imageinfo = ImageInformation(metaDict)
+    header = Header(metaDict)
+    burstinfo = Burst(metaDict,
+        swathtiming.linesPerBurst,
+        imageinfo.azimuthFrequency,
+        header.startTime)
+
+    metadata = MetaDataSentinel1(header,
+        ProductInformation(metaDict),
+        imageinfo,
+        swathtiming,
+        burstinfo,
+        GeolocationGrid(metaDict))
+
+    return metadata
+end
+
+
+
+
+
+""""
+Header
+
+    Constucture for the Header structure. 
+
+    It takes a dictionary containing the full sentinel-1 swath metadata and extracts the Header as a structure. Input in the header file:
+        missionId,
+        productType,
+        polarisation,
+        missionDataTakeId,
+        swath,
+        mode,
+        startTime,
+        stopTime,
+        aqusitionTime,
+        absoluteOrbitNumber,
+        imageNumber
+
+    Input:
+        metaDict[dict]: a dictionary of the metadata.
+
+    output:
+        Header[structure of Header]
+    
+"""
+function Header(metaDict)::Header
```

Comment:
Please add input type everywhere.  Here `metaDict::Dict`. The types should be as broad as what makes sense for the function to work. E.g. the abstract type `::Number` is often used for function that works with all sorts of number. Number includes Int64, Floar32, UInt8, Complex{Float64} and many more

It is good for a couple of reasons.
1. Adding types makes it easy to use Julia's multiple dispatch features
2. It gives good error messages when functions are used wrong.